agent/agent.go

```diff
@@ -207,6 +207,16 @@ LOOP:
 			grip.Info("agent loop canceled")
 			return nil
 		case <-timer.C:
+			// Check the cedar GRPC connection so we can fail early
+			// and avoid task system failures.
+			err := utility.Retry(ctx, func() (bool, error) {
+				_, err := a.comm.GetCedarGRPCConn(ctx)
+				return true, err
+			}, utility.RetryOptions{MaxAttempts: 20, MaxDelay: time.Minute})
```

Comment:
Can we reduce the number of retries and keep the max delay short? The longer it spends trying to connect to Cedar, the more likely the app server is to terminate the host because it thinks the agent is dead since it's not sending any requests.